### PR TITLE
Cleaned and linted the source code for the vesselness filter.

### PR DESCRIPTION
**Pull request by biocubed**
Please evaluate the code for completeness, and comments

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -16,37 +18,47 @@ find_package(catkin REQUIRED COMPONENTS
   dynamic_reconfigure
 )
 
-
+#Add messages
```
Comment (lucbettaieb) [inline CMakeLists.txt]: space after comment

File: CMakeLists.txt [hunk 2]
```diff
@@ -16,37 +18,47 @@ find_package(catkin REQUIRED COMPONENTS
...
 )
 
-#
 generate_dynamic_reconfigure_options(
   cfg/vesselness_params_.cfg
   #...
 )
 
+
+# package definition...
```
Comment (lucbettaieb) [inline CMakeLists.txt]: Format all these comments the same way

File: CMakeLists.txt [hunk 3]
```diff
@@ -154,55 +179,54 @@ target_link_libraries(vesselness_image_filter_cpu_bw_node
...
 
-target_link_libraries(vesselness_viewer_gpu_node
-   vesselness_viewer_gpu_lib
-   vesselness_image_filter_common_lib
-   ${CUDA_LIBRARIES}
-   ${OpenCV_LIBRARIES}
-   ${catkin_LIBRARIES}
-)
+  target_link_libraries(vesselness_image_filter_gpu_node
+    vesselness_image_filter_common_lib
```
Comment (lucbettaieb) [inline CMakeLists.txt:195]: make sure all this spacing is a-ok

File: CMakeLists.txt [hunk 4]
```diff
@@ -55,83 +67,97 @@ include_directories(
...
 
-
-
-target_link_libraries(vesselness_image_filter_cpu_lib
-  vesselness_image_filter_common_lib
-  ${OpenCV_LIBRARIES}
-  ${catkin_LIBRARIES}
- )
+# cpu based filter executable:
+ADD_EXECUTABLE(vesselness_image_filter_cpu_node
```
Comment (lucbettaieb) [inline CMakeLists.txt]: caps?

File: CMakeLists.txt [hunk 5]
```diff
@@ -141,11 +167,10 @@ target_link_libraries(vesselness_image_filter_cpu_node
  )
 
 
-target_link_libraries(vesselness_image_filter_bw_cpu_lib
-    vesselness_image_filter_common_lib
-    ${OpenCV_LIBRARIES}
-    ${catkin_LIBRARIES}
-    )
+# B & W filter executable.
+ADD_EXECUTABLE(vesselness_image_filter_cpu_bw_node
```
Comment (lucbettaieb) [inline CMakeLists.txt]: more caps?

File: CMakeLists.txt [hunk 6]
```diff
@@ -154,55 +179,54 @@ target_link_libraries(vesselness_image_filter_cpu_bw_node
    ${catkin_LIBRARIES}
  )
 
-target_link_libraries(vesselness_lib
-   ${OpenCV_LIBRARIES}
-   ${catkin_LIBRARIES}
- )
+
+# CPU viewer executable:
+ADD_EXECUTABLE(vesselness_image_filter_cpu_viewer_node
```
Comment (lucbettaieb) [inline CMakeLists.txt]: ?

File: include/vesselness_image_filter_cpu/vesselness_filter_node_cpu.h [hunk 7]
```diff
@@ -35,94 +35,123 @@
  *
  */
 
-#ifndef IMAGESEGMENTCPUH
-#define IMAGESEGMENTCPUH
+#ifndef VESSELNESS_IMAGE_FILTER_NODE_CPU_H
+#define VESSELNESS_IMAGE_FILTER_NODE_CPU_H
 
 
 #include <vesselness_image_filter_common/vesselness_image_filter_common.h>
```
Comment (lucbettaieb) [inline include/vesselness_image_filter_cpu/vesselness_filter_node_cpu.h]: whitespace

File: include/vesselness_image_filter_cpu/vesselness_filter_node_cpu.h [hunk 8]
```diff
@@ -35,94 +35,123 @@
...
-    void segmentImage(const cv::Mat &, cv::Mat &);
-
-    
-    //Update object parameters.
-    void updateKernels(const segmentThinParam &);
-	
+  /**
+   * @brief the grey input image.
+   */
+  cv::Mat greyImage;
```
Comment (lucbettaieb) [inline include/vesselness_image_filter_cpu/vesselness_filter_node_cpu.h]: for class variables i usually put a trailing _

File: include/vesselness_image_filter_cpu/vesselness_filter_node_cpu.h [hunk 9]
```diff
@@ -35,94 +35,123 @@
...
+
+  /**
+   * @brief the image filtering mask.
+   */
+  cv::Mat imageMask;
+
+  /**
+   * @brief initialize the required gaussian kernels.
+   */
+  void  initKernels() override;
```
Comment (lucbettaieb) [inline include/vesselness_image_filter_cpu/vesselness_filter_node_cpu.h]: 2 spaces after void

File: include/vesselness_image_filter_cpu/vesselness_filter_node_cpu_bw.h [hunk 10]
```diff
@@ -35,95 +35,127 @@
  *
  */
 
-#ifndef IMAGESEGMENTCPUH
-#define IMAGESEGMENTCPUH
-
+#ifndef VESSELNESS_IMAGE_FILTER_NODE_CPU_BW_H
+#define VESSELNESS_IMAGE_FILTER_NODE_CPU_BW_H
```
Comment (lucbettaieb) [inline include/vesselness_image_filter_cpu/vesselness_filter_node_cpu_bw.h]: make sure this whitespace is good

File: include/vesselness_image_filter_cpu/vesselness_filter_node_cpu_bw.h [hunk 11]
```diff
@@ -35,95 +35,127 @@
...
+/**
+ * @brief The BW vesselness node class which extends the vesselness base class.
+ */
+class VesselnessNodeCPUBW: public VesselnessNodeBase
+{
+private:  
+  /**
+   * @brief the grey input image.
+   */
+  cv::Mat greyImage;
```
Comment (lucbettaieb) [inline include/vesselness_image_filter_cpu/vesselness_filter_node_cpu_bw.h]: _'s if you so choose.  i would.

File: include/vesselness_image_filter_cpu/vesselness_lib.h [hunk 12]
```diff
@@ -38,22 +38,52 @@
 
 
 
-#ifndef VESSELNESSCPULIBH
-#define VESSELNESSCPULIBH
+#ifndef VESSELNESS_LIB_H
```
Comment (lucbettaieb) [inline include/vesselness_image_filter_cpu/vesselness_lib.h]: VESSELNESS_IMAGE_FILTER_CPU_VESSELNESS_LIB_H

File: include/vesselness_image_filter_cpu/vesselness_filter_node_cpu.h [hunk 13]
```diff
@@ -35,94 +35,123 @@
  *
  */
 
-#ifndef IMAGESEGMENTCPUH
-#define IMAGESEGMENTCPUH
+#ifndef VESSELNESS_IMAGE_FILTER_NODE_CPU_H
```
Comment (lucbettaieb) [inline include/vesselness_image_filter_cpu/vesselness_filter_node_cpu.h]: VESSELNESS_IMAGE_FILTER_CPU_VESSELNESS_FILTER_NODE_CPU_H

File: include/vesselness_image_filter_cpu/vesselness_lib.h (hunk 12 above)
Comment (lucbettaieb) [reply include/vesselness_image_filter_cpu/vesselness_lib.h]: generally is "PACKAGE_NAME_CLASS_NAME_H"

File: src/cpu/vesselness_image_filter_cpu_bw.cpp [hunk 14]
```diff
@@ -45,97 +45,102 @@
 
 #include <vesselness_image_filter_cpu/vesselness_filter_node_cpu_bw.h>
 
-using namespace cv;
-
+// deallocates the memory.
```
Comment (lucbettaieb) [inline src/cpu/vesselness_image_filter_cpu_bw.cpp]: remove all unnecessary/obvious comments

File: src/cpu/vesselness_image_filter_cpu_bw.cpp [hunk 15]
```diff
@@ -45,97 +45,102 @@
...
 void VesselnessNodeCPUBW::deallocateMem()
```
Comment (lucbettaieb) [inline src/cpu/vesselness_image_filter_cpu_bw.cpp]: if this does nothing get rid of it

File: src/cpu/vesselness_image_filter_cpu_bw.cpp [hunk 16]
```diff
@@ -45,97 +45,102 @@
...
 {
-
-
 }
 
-VesselnessNodeCPUBW::VesselnessNodeCPUBW(const char* subscriptionChar,const char* publicationChar):VesselnessNodeBase(subscriptionChar,publicationChar)
+// constructor.
```
Comment (lucbettaieb) [inline src/cpu/vesselness_image_filter_cpu_bw.cpp]: remove not-needed comments

File: src/cpu/vesselness_image_filter_cpu_bw.cpp [hunk 17]
```diff
@@ -45,97 +45,102 @@
...
+  for (int ix = -kSizeEnd; ix < kSizeEnd+1; ix++)
+  {
+    for (int iy = -kSizeEnd; iy < kSizeEnd+1; iy++)
+    {
+      float ixD = static_cast<float>  (ix);
+      float iyD = static_cast<float>  (iy);
 
-            gaussKernel_XX.at<float>(iy+kSizeEnd,ix+kSizeEnd) = (ixD*ixD)/(var*var)*gaussFnc(var,ixD,iyD)-1/(var)*gaussFnc(var,ixD,iyD);
+      gaussKernel_XX.at<float>(iy+kSizeEnd, ix+kSizeEnd) =
+        (ixD*ixD)/(var*var)*gaussFnc(var, ixD, iyD)-1/(var)*gaussFnc(var, ixD, iyD);
```
Comment (lucbettaieb) [inline src/cpu/vesselness_image_filter_cpu_bw.cpp]: comments on math are usually helpful

File: src/cpu/vesselness_image_filter_cpu_bw.cpp [hunk 18]
```diff
@@ -45,97 +45,102 @@
...
+{
+  float betaParam(filterParameters.betaParam);
+  float cParam(filterParameters.cParam);
 
+  // Actual process segmentation code:
+  cv::cvtColor(src, greyImage, CV_BGR2GRAY);
+  greyImage.convertTo(greyFloat, CV_32FC1, 1.0, 0.0);
 
-void  VesselnessNodeCPUBW::segmentImage(const Mat& src,Mat& dst) {
+  // @TODO Replace the C style pointer casting with the reinterpret_cast
```
Comment (lucbettaieb) [inline src/cpu/vesselness_image_filter_cpu_bw.cpp]: // TODO(biocubed): comment

File: src/cpu/vesselness_image_filter_cpu_bw.cpp [hunk 19]
```diff
@@ -45,97 +45,102 @@
...
 
-    float betaParam(filterParameters.betaParam);
-    float cParam(filterParameters.cParam);
-    //Actual process segmentation code:
-    cvtColor(src,greyImage,CV_BGR2GRAY);
-    greyImage.convertTo(greyFloat,CV_32FC1,1.0,0.0);
+  float *greyFloatPtr = reinterpret_cast<float*> (greyFloat.data);
```
Comment (lucbettaieb) [inline src/cpu/vesselness_image_filter_cpu_bw.cpp]: If you can help it, avoid raw pointers... but idk what you're doing so go nuts

File: src/cpu/vesselness_image_filter_cpu_bw.cpp [hunk 20]
```diff
@@ -45,97 +45,102 @@
...
 
 
-    std::cout << "at while loop" << std::endl;
+  std::cout << "at while loop" << std::endl;
 
-    //From Frangi et al.
-    //for each image, evaluate its eigen vectors, then look at the cost
+  //From Frangi et al.
+  //for each image, evaluate its eigen vectors, then look at the cost
 	for(int i =0 ; i < pixCount; i++){
```
Comment (lucbettaieb) [inline src/cpu/vesselness_image_filter_cpu_bw.cpp]: this is weird formatting, pls fix

File: src/cpu/vesselness_image_filter_cpu_bw.cpp [hunk 21]
```diff
@@ -213,26 +218,20 @@ void  VesselnessNodeCPUBW::segmentImage(const Mat& src,Mat& dst) {
...
-
-//destructor function
-VesselnessNodeCPUBW::~VesselnessNodeCPUBW(){
-    //clean up the Mats and memory
-
-
-
+VesselnessNodeCPUBW::~VesselnessNodeCPUBW()
+{
+  // clean up the Mats and memory (not necessary)
```
Comment (lucbettaieb) [inline src/cpu/vesselness_image_filter_cpu_bw.cpp]: remove unnecessary comments

File: src/cpu/vesselness_image_filter_cpu_node.cpp [hunk 22]
```diff
@@ -46,19 +46,17 @@
 #include <vesselness_image_filter_cpu/vesselness_filter_node_cpu.h>
 
 
-//This file instantiates a node object of the vesselness cpu node.
+// This file instantiates a node object of the vesselness cpu node.
 int main(int argc, char** argv)
 {
-
+    // initialize the object.
```
Comment (lucbettaieb) [inline src/cpu/vesselness_image_filter_cpu_node.cpp]: dont need this

File: src/cpu/vesselness_image_viewer_node.cpp [hunk 23]
```diff
@@ -42,7 +42,7 @@
 #include <opencv2/imgproc/imgproc.hpp>
 #include <opencv2/highgui/highgui.hpp>
 #include <vesselness_image_filter_cpu/vesselness_lib.h>
-using namespace cv;
+
```
Comment (lucbettaieb) [inline src/cpu/vesselness_image_viewer_node.cpp]: whitespace

File: src/cpu/vesselness_image_viewer_node.cpp [hunk 24]
```diff
@@ -51,23 +51,28 @@ class ImageConverter
...
 public:
   ImageConverter()
-    : it_(nh_)
+    :it_(nh_)
   {
-    // Subscrive to input video feed and publish output video feed
-    image_sub_ = it_.subscribe("image_thin", 1, 
+    // Subscribe to input video feed and display output video feed
+    // use the image callback to display the image.
+    // @TODO understand how using opencv HIGHGUI with  libQT vs. libGTK makes a difference.
```
Comment (lucbettaieb) [inline src/cpu/vesselness_image_viewer_node.cpp]: todo style

File: src/gpu/cpp/vesselness_image_viewer_gpu_node.cpp [hunk 25]
```diff
@@ -43,34 +43,38 @@
 #include <opencv2/highgui/highgui.hpp>
 #include "vesselness_image_filter_gpu/vesselness_viewer_kernels.h"
 
-using namespace cv;
-//Converts a simgle image into a displayable RGB format.
-
-static const std::string OPENCV_WINDOW = "Image window";
 
+//Converts a single image into a displayable RGB format.
```
Comment (lucbettaieb) [inline src/gpu/cpp/vesselness_image_viewer_gpu_node.cpp]: style

File: src/gpu/cu/vesselness_image_filter_kernels.cu [hunk 26]
```diff
@@ -37,21 +37,24 @@
...
+// This file defines the kernel functions used by the thin segmentation cuda code.
+// currently this file is UNUSED.
+// @TODO use this file to define cuda kernel functions code readability.
 
-using namespace cv::cuda;
 
 // Need to remove this #define, replace it with a __local__ cuda function.
 // #define gaussFncGPU(var,x,y) 1.0f/(3.1415f*2.0f*var)*((float) ));
 
+/*
```
Comment (lucbettaieb) [inline src/gpu/cu/vesselness_image_filter_kernels.cu]: dont include commented code in any kind of release-worthy thing

File: src/gpu/cu/vesselness_image_filter_kernels.cu [hunk 27]
```diff
@@ -37,21 +37,24 @@
...
 
 
+
 #include "vesselness_image_filter_gpu/vesselness_image_filter_kernels.h"
 
 
-//This file defines the kernel functions used by the thin segmentation cuda code.
+// This file defines the kernel functions used by the thin segmentation cuda code.
+// currently this file is UNUSED.
+// @TODO use this file to define cuda kernel functions code readability.
```
Comment (lucbettaieb) [inline src/gpu/cu/vesselness_image_filter_kernels.cu]: todo fix

File: src/gpu/cu/vesselness_viewer_kernel.cu [hunk 28]
```diff
@@ -37,64 +37,68 @@
...
+  srcG.upload(srcMatLocked, streamInfo);
+  cv::cuda::GpuMat scaledSrc, scaledSrcU;
+
+  // scale the image from [0-3.141] x [0-1] to [0-255] x [0 - 255].
+  cv::cuda::multiply(srcG, cv::Scalar(179/3.14159, 255.0), scaledSrc, 1, CV_32FC2, streamInfo);
+
+  // convert the image from a floating point image
+  // to an 8 bit unsigned char image.
+  scaledSrc.convertTo(scaledSrcU, CV_8UC2, streamInfo);
+
```
Comment (lucbettaieb) [inline src/gpu/cu/vesselness_viewer_kernel.cu:65]: whitespace

File: src/vesselness_image_filter_common.cpp [hunk 29]
```diff
@@ -43,120 +43,118 @@
...
+  {
+    image_pub_.publish(cv_Out.toImageMsg());
+    ROS_INFO("published new image");
+  }
 }
 
 
-
-
+// This is the blank destructor.
```
Comment (lucbettaieb) [inline src/vesselness_image_filter_common.cpp]: dont need this

**Review by lucbettaieb — DISMISSED** [review]
You should run the ROS linter on your code using `roslint`, or if you want to ignore BS things, you can use the fetch linter by installing `fetch_tools`.  Their linter is the ROS linter without the BS.  https://github.com/fetchrobotics/fetch_tools

After that's installed you just need to type `fetch lint /path/to/lintable/files`

Other than that, I'd recommend removing unnecessary comments and commented code.  Check my specific comments for more detail.

File: CMakeLists.txt (hunk 1 above)
Comment (biocubed) [reply CMakeLists.txt]: fixed

File: CMakeLists.txt (hunk 2 above)
Comment (biocubed) [reply CMakeLists.txt]: fixed

File: CMakeLists.txt (hunk 4 above)
Comment (biocubed) [reply CMakeLists.txt]: fixed

File: CMakeLists.txt (hunk 5 above)
Comment (biocubed) [reply CMakeLists.txt]: more fixed

File: CMakeLists.txt (hunk 6 above)
Comment (biocubed) [reply CMakeLists.txt]: fixed ?

File: CMakeLists.txt (hunk 3 above)
Comment (biocubed) [reply CMakeLists.txt:195]: Fixed

File: include/vesselness_image_filter_cpu/vesselness_filter_node_cpu.h (hunk 13 above)
Comment (biocubed) [reply include/vesselness_image_filter_cpu/vesselness_filter_node_cpu.h]: fixed

File: include/vesselness_image_filter_cpu/vesselness_filter_node_cpu.h (hunk 7 above)
Comment (biocubed) [reply include/vesselness_image_filter_cpu/vesselness_filter_node_cpu.h]: fixed

File: src/gpu/cu/vesselness_viewer_kernel.cu (hunk 28 above)
Comment (biocubed) [reply src/gpu/cu/vesselness_viewer_kernel.cu:65]: fixed

**Pull request merged**